core/chain-configs/src/genesis_config.rs

```diff
@@ -72,88 +70,143 @@ pub struct GenesisConfig {
     /// Maximum inflation on the total supply every epoch (this is a number between 0 and 100)
     pub max_inflation_rate: u8,
     /// Total supply of tokens at genesis.
-    pub total_supply: u128,
+    #[serde(with = "u128_dec_format", skip_deserializing)]
+    pub total_supply: Balance,
     /// Expected number of blocks per year
-    pub num_blocks_per_year: u64,
+    pub num_blocks_per_year: NumBlocks,
     /// Protocol treasury account
     pub protocol_treasury_account: AccountId,
     /// Fishermen stake threshold.
     #[serde(with = "u128_dec_format")]
     pub fishermen_threshold: Balance,
 }
 
+/// Records in storage at genesis (get split into shards at genesis creation).
+#[derive(
+    Debug,
+    Clone,
+    SmartDefault,
+    derive_more::AsRef,
+    derive_more::AsMut,
+    derive_more::From,
+    Serialize,
+    Deserialize,
+)]
+pub struct GenesisRecords(Vec<StateRecord>);
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct Genesis {
+    #[serde(flatten)]
+    pub config: GenesisConfig,
+    pub records: GenesisRecords,
+    phantom: PhantomData<()>,
```

Comment:
It is to prevent manual construction of this struct (as we need to have total supply computed, which used to be part of the external contact, and now it is moved to `new`, so you cannot forget to call it)

Reply:
Nice! Can you add a comment about this?